matplotlib/figure.pyi

```diff
@@ -72,6 +72,7 @@ class FigureBase(Artist):
     def add_axes(self, *args, **kwargs) -> Axes: ...
     @overload
     def add_subplot(self, projection: Literal['3d'], *args, **kwargs) -> Axes3D: ...
+    @overload
```

Comment:
@maflAT, please take a quick look at https://github.com/microsoft/python-type-stubs/blob/main/tests/matplotlib/pyplot_tests.py and then add a new `tests/matplotlib/figure_tests.py` file that does similar `assert_type` verifications of the `add_subplot` overloads.

We'll build up these tests over time as a way to prevent regressions of this sort.

Reply:
@debonte, I added a few tests.
While doing so i found another bug in the Axes3D signature (`projection `should be keyword-only).

I could not not get the Axes3D tests to work. For some reason pyright thinks Axes3D is a Callable and not a Type. I don't know how to fix this.

```
C:\_WIP_\python-type-stubs\tests>pyright --lib --pythonversion 3.11 -p ../pyrighttestconfig.json .
The --lib option is deprecated. Pyright now defaults to using library code to infer types.
Loading configuration file at C:\_WIP_\python-type-stubs\pyrighttestconfig.json
Assuming Python platform Windows
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
Searching for source files
Found 3 source files
pyright 1.1.303
C:\_WIP_\python-type-stubs\tests\matplotlib\figure_tests.py
  C:\_WIP_\python-type-stubs\tests\matplotlib\figure_tests.py:35:17 - error: Expected type expression but received "(...) -> Unknown" (reportGeneralTypeIssues)
  C:\_WIP_\python-type-stubs\tests\matplotlib\figure_tests.py:38:17 - error: Expected type expression but received "(...) -> Unknown" (reportGeneralTypeIssues)
  C:\_WIP_\python-type-stubs\tests\matplotlib\figure_tests.py:41:17 - error: Expected type expression but received "(...) -> Unknown" (reportGeneralTypeIssues)
  C:\_WIP_\python-type-stubs\tests\matplotlib\figure_tests.py:44:17 - error: Expected type expression but received "(...) -> Unknown" (reportGeneralTypeIssues)
4 errors, 0 warnings, 0 informations
Completed in 0.998sec
```

Reply:
We don't currently have stubs for `mpl_toolkits` and this behavior is somehow caused by the decorators that are applied to `Axes3D` in the implementation. I appreciate your efforts here, but would suggest that we skip the 3D tests for now. The rest of your changes look good!

```python
@_docstring.interpd
@_api.define_aliases({
    "xlim": ["xlim3d"], "ylim": ["ylim3d"], "zlim": ["zlim3d"]})
class Axes3D(Axes):
```